### PR TITLE
Comet Integration

### PR DESCRIPTION
**Pull request by DN6**
## Before submitting (checklist)

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contribution guide](https://github.com/catalyst-team/catalyst/blob/master/CONTRIBUTING.md)?
- [x] Did you check the code style? `catalyst-make-codestyle && catalyst-check-codestyle` (`pip install -U catalyst-codestyle`).
- [x] Did you make sure to update the docs? We use Google format for all the methods and classes.
- [x] Did you check the docs with `make check-docs`?
- [ ] Did you write any new necessary tests?
- [x] Did you check that your code passes the unit tests `pytest .` ? 
- [x] Did you add your new functionality to the docs?
- [ ] Did you update the [CHANGELOG](https://github.com/catalyst-team/catalyst/blob/master/CHANGELOG.md)?
- [x] Did you run [colab minimal CI/CD](https://colab.research.google.com/github/catalyst-team/catalyst/blob/master/examples/notebooks/colab_ci_cd.ipynb) with `latest` and `minimal` requirements?
- [x] Did you check XLA integration with [single](https://colab.research.google.com/github/catalyst-team/catalyst/blob/master/examples/notebooks/Catalyst_XLA_single_process.ipynb) and [multiple]((https://colab.research.google.com/github/catalyst-team/catalyst/blob/master/examples/notebooks/Catalyst_XLA_multi_process.ipynb)) processes?

## Description

This PR adds the option to log metrics, parameters and any other type of data asset to [Comet](https://www.comet.ml/site/data-scientists/)

## Type of Change

- [ ] Examples / docs / tutorials / contributors update
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves an existing feature)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

### FAQ
Please review the FAQ before submitting an issue:
- [x] I have read the [documentation and FAQ](https://catalyst-team.github.io/catalyst/)
- [x] I have reviewed the [minimal examples section](https://github.com/catalyst-team/catalyst#minimal-examples)
- [x] I have checked the [changelog](https://github.com/catalyst-team/catalyst/blob/master/CHANGELOG.md) for main framework updates
- [x] I have read the [contribution guide](https://github.com/catalyst-team/catalyst/blob/master/CONTRIBUTING.md)
- [x] I have joined [Catalyst slack (#__questions channel)](https://join.slack.com/t/catalyst-team-core/shared_invite/zt-d9miirnn-z86oKDzFMKlMG4fgFdZafw) for issue discussion

### CODE REVIEW

**Comment by DN6** [comment]
Hi @Scitator. I think I've addressed all the code style issues, run the pre-commit hooks, and the docs tests. They all seem to be passing. I'm still not able to see the CometLogger when I render the docs page though. Let me know if I missed something.

File: catalyst/loggers/comet.py [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+                    )
+                }
+            # ...
+
+        runner = CustomRunner().run()
+
+    Config API example:
+    .. code-block:: yaml
+        loggers:
+            neptune:
```
Comment (Podidiving) [inline catalyst/loggers/comet.py]: Forgot to rename it

File: catalyst/loggers/comet.py [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+                _target_: CometLogger
+                project_name: my_project
+        ...
+    Hydra API example:
+    .. code-block:: yaml
+        loggers:
+            neptune:
```
Comment (Podidiving) [inline catalyst/loggers/comet.py]: Forgot to rename it

File: catalyst/loggers/comet.py [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+    .. code-block:: python
+
+        from catalyst import dl
+
+        runner = dl.SupervisedRunner()
+        runner.train(
+            ...
+            loggers={
+                "comet": dl.CometLogger(
+                    project_name="my-comet-project
```
Comment (Podidiving) [inline catalyst/loggers/comet.py]: missing "

File: catalyst/loggers/comet.py [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+
+    def __init__(
+        self,
+        workspace: str = None,
+        project_name: str = None,
+        experiment_id: str = None,
+        comet_mode: str = "online",
+        tags: List = None,
+        logging_frequency: int = 1,
+        **experiment_kwargs,
```
Comment (Podidiving) [inline catalyst/loggers/comet.py]: mb add `api_key` as an explicit parameter?

**Comment by Scitator** [comment]
Hi @DN6 ,
For the docs support, you should also update https://github.com/catalyst-team/catalyst/blob/master/docs/api/loggers.rst ;)

File: catalyst/loggers/comet.py (hunk 4 above)
Comment (DN6) [reply catalyst/loggers/comet.py]: Hi @Podidiving, generally we recommend users set their API Key either as an environment variable or in a `comet.config` file on their machine.

**Comment by Scitator** [comment]
waiting for the docs and codestyle and ready to merge 🚀

**Comment by DN6** [comment]
@Scitator  @Podidiving I've cleaned up the docstrings, added the Comet Logger information to the docs and I run the codestyle pre-commit hooks again. 

Additionally, we've prepared a tutorial Colab Notebook for anyone to test out the integration [here](https://colab.research.google.com/drive/1TaG27HcMh2jyRKBGsqRXLiGUfsHVyCq6?usp=sharing). Would it be possible to include it somewhere in the README (Maybe under [Examples, Notebooks, and Starter Kits](https://github.com/catalyst-team/catalyst#examples-notebooks-and-starter-kits))?

**Comment by Scitator** [comment]
@DN6 could you please check the codestyle?

**Comment by DN6** [comment]
@Scitator Done. Checked with the Colab CI/CD tool and it seems like it's passing. Output from 
`! cat ./catalyst/codestyle.txt`

```
Skipped 58 files
0
``` 
For the example notebook, should I open up a separate PR or add to this one?

**Pull request merged**